tripy/tripy/frontend/trace/ops/slice.py

```diff
@@ -256,8 +256,36 @@ def clamp_bound(bound: Union[int, Tensor]) -> Union[int, Tensor]:
     return out
 
 
-# Because the helper is called inside another function, we need to skip one entry in the call stack to find
-# the original call to user code.
-@frontend_utils.convert_to_tensors(skip_num_stack_entries=1)
+@frontend_utils.convert_to_tensors()
 def slice_helper(tensor, *slice_params: TensorLike):
```

Comment:
I'm wondering if it would be better to do the conversion to tensors and the stack info in the caller? The reason `skip_num_stack_entries` is unreliable is that the signature of the API can differ from that of the decorated function, which breaks our AST parsing. By still having the helper, it seems like we would have the same problem.

Reply:
Have you run into a case where that's actually happened? The error messages in the test cases generate correctly.

We could indeed get rid of the decorator altogether and handle everything directly in `__getitem__`. The reason for having the helper is that the decorator normally would not be able to handle `slice` inputs and we would have to preprocess them. In any case, we do need to know which inputs are `slice`s and which ones are `int`s to determine things like distinguishing between a slice of size 1 (returns a list of length 1) and a single index (returns a single entry).

Reply:
In this case, the code that would be parsed would be the user's subscripted tensor, i.e., the subscript AST node, whose signature would not match that of our override for `__getitem__` regardless. We could pull in the portions of the code from the decorator that convert values into tensors, but it would not be much different in principle from what the decorator is already doing.

Reply:
> Have you run into a case where that's actually happened? 

Yes, but not in slice. I made some fixes, but we used to have some APIs that used decorated helpers with different arguments than the public facing functions. The stack information would point to completely wrong things in those cases. I'm fine with keeping the decorator if it works fine for slice and as long as our testing for it is robust (which I think it is).

Reply:
Thanks for the information. Definitely a good reason to have detailed tests for error cases.